### ISSUE TITLE
라벨인코딩 데이터 성능 비교(LR, DTR, RFR)

### ISSUE THREAD

**Issue by UGeunJi** [SeokJuGo/DataAnalysis_PredictTrafficJeju#2]
<html>
<body>
<pre><code class="fenced-code-block language-python">import random
import os
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import scipy.stats as stats
%matplotlib inline
import seaborn as sns
import warnings
import plotly.express as px</code></pre>
<pre><code class="fenced-code-block language-python">train = pd.read_csv(&#39;Dataset/train.csv&#39;)
test = pd.read_csv(&#39;Dataset/test.csv&#39;)</code></pre>
<pre><code class="fenced-code-block language-python">from sklearn.preprocessing import LabelEncoder

items =  [&#39;day_of_week&#39;, &#39;road_name&#39;, &#39;start_node_name&#39;, &#39;start_turn_restricted&#39;, &#39;end_node_name&#39;, &#39;end_turn_restricted&#39;]
train_enco = train.copy()
encoder = LabelEncoder()

for i in items:
    encoder = LabelEncoder()
    encoder.fit(train_enco[i])
    train_enco[i] = encoder.transform(train_enco[i])

train_enco</code></pre>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

<pre><code class="fenced-code-block">.dataframe tbody tr th {
    vertical-align: top;
}

.dataframe thead th {
    text-align: right;
}</code></pre>
<p></style></p>

  | id | base_date | day_of_week | base_hour | lane_count | road_rating | road_name | multi_linked | connect_code | maximum_speed_limit | ... | road_type | start_node_name | start_latitude | start_longitude | start_turn_restricted | end_node_name | end_latitude | end_longitude | end_turn_restricted | target
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
TRAIN_0000000 | 20220623 | 1 | 17 | 1 | 106 | 46 | 0 | 0 | 60.0 | ... | 3 | 386 | 33.427747 | 126.662612 | 0 | 386 | 33.427749 | 126.662335 | 0 | 52.0
TRAIN_0000001 | 20220728 | 1 | 21 | 2 | 103 | 34 | 0 | 0 | 60.0 | ... | 0 | 32 | 33.500730 | 126.529107 | 1 | 2 | 33.504811 | 126.526240 | 0 | 30.0
TRAIN_0000002 | 20211010 | 4 | 7 | 2 | 103 | 36 | 0 | 0 | 80.0 | ... | 0 | 426 | 33.279145 | 126.368598 | 0 | 179 | 33.280072 | 126.362147 | 0 | 61.0
TRAIN_0000003 | 20220311 | 0 | 13 | 2 | 107 | 57 | 0 | 0 | 50.0 | ... | 0 | 58 | 33.246081 | 126.567204 | 0 | 203 | 33.245565 | 126.566228 | 0 | 20.0
TRAIN_0000004 | 20211005 | 6 | 8 | 2 | 103 | 35 | 0 | 0 | 80.0 | ... | 0 | 267 | 33.462214 | 126.326551 | 0 | 269 | 33.462677 | 126.330152 | 0 | 38.0
... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ...
TRAIN_4701212 | 20211104 | 1 | 16 | 1 | 107 | 0 | 0 | 0 | 50.0 | ... | 0 | 79 | 33.422145 | 126.278125 | 0 | 42 | 33.420955 | 126.273750 | 0 | 20.0
TRAIN_4701213 | 20220331 | 1 | 2 | 2 | 107 | 0 | 0 | 0 | 80.0 | ... | 3 | 31 | 33.472505 | 126.424368 | 0 | 31 | 33.472525 | 126.424890 | 0 | 65.0
TRAIN_4701214 | 20220613 | 3 | 22 | 2 | 103 | 35 | 0 | 0 | 60.0 | ... | 0 | 19 | 33.447183 | 126.912579 | 0 | 214 | 33.444121 | 126.912948 | 0 | 30.0
TRAIN_4701215 | 20211020 | 2 | 2 | 2 | 103 | 37 | 0 | 0 | 80.0 | ... | 0 | 392 | 33.443596 | 126.431817 | 0 | 23 | 33.444996 | 126.433332 | 0 | 73.0
TRAIN_4701216 | 20211019 | 6 | 6 | 2 | 107 | 1 | 0 | 0 | 60.0 | ... | 0 | 190 | 33.256785 | 126.508940 | 0 | 247 | 33.257130 | 126.510364 | 0 | 35.0

<p>4701217 rows × 23 columns</p>
</div>

<pre><code class="fenced-code-block language-python">from sklearn.model_selection import train_test_split</code></pre>
<pre><code class="fenced-code-block language-python">X = train_enco.drop([&#39;id&#39;,&#39;multi_linked&#39;, &#39;target&#39;,&#39;height_restricted&#39;, &#39;connect_code&#39;, &#39;vehicle_restricted&#39;], axis=1)
y = train_enco[&#39;target&#39;] 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 42)</code></pre>
<pre><code class="fenced-code-block language-python">print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)</code></pre>
<pre><code class="fenced-code-block">(3760973, 17) (940244, 17) (3760973,) (940244,)</code></pre>
<pre><code class="fenced-code-block language-python">from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LinearRegression

lin_reg = LinearRegression()

scores = cross_val_score(lin_reg, X_train, y_train, scoring = &#39;neg_mean_squared_error&#39;, cv = 5)
lin_reg_rmse = np.sqrt((-scores).mean())
lin_reg_rmse</code></pre>
<pre><code class="fenced-code-block">12.598237553430662</code></pre>
<pre><code class="fenced-code-block language-python">from sklearn.tree import DecisionTreeRegressor

tree_reg = DecisionTreeRegressor()

tree_scores = cross_val_score(tree_reg, X_train, y_train, scoring=&quot;neg_mean_squared_error&quot;, cv=10, n_jobs=-1)

tree_rmse = np.sqrt(-tree_scores).mean()
tree_rmse</code></pre>
<pre><code class="fenced-code-block">5.633164366248218</code></pre>
<pre><code class="fenced-code-block language-python">from sklearn.ensemble import RandomForestRegressor

rf_reg = RandomForestRegressor()

rf_scores = cross_val_score(rf_reg, X_train, y_train, scoring=&quot;neg_mean_squared_error&quot;, cv=2, n_jobs=-1)

rf_rmse = np.sqrt(-rf_scores).mean()
rf_rmse</code></pre>
<pre><code class="fenced-code-block">4.47978887708306</code></pre>

</body>
</html>
